docs/concepts/crds/task.md

```diff
@@ -0,0 +1,97 @@
+      value: /tmp/output
+    - name: secretVolume # specifies where backend secret has been mounted
+      value: secret-volume
+  - name: update-status
+    params:
+    - name: outputDir # specifies where the previous step wrote the output file. it will read that file and update the status of respective resources accordingly.
+      value: /tmp/output
+  volumes:
+  - name: secret-volume
+    secret:
+      secretName: ${REPOSITORY_SECRET_NAME}
+```
+
+This `Task` uses two functions to backup a PostgreSQL database. The first step indicates `pg-backup` function that dumps PostgreSQL database and uploads the dumped file. The second step indicates `update-status` function which updates the status of the `BackupSession` and `Repository` crd for respective backup.
+
+Here, we are going to describe the various sections of a `Task` crd.
+
+### Task `Spec`
+
+A `Task` object has the following fields in the `spec` section:
+
+#### spec.steps
+
+`spec.steps` section specifies a list of functions and their parameters in the order they should be executed. You can also template this section using the [variables](/docs/concepts/crds/functions.md#stash-provided-variables) that Stash can resolve itself. Stash will resolve all the variables and create a pod definition with the container specification specified in the respective `Function` specified in `steps` section.
+
+Each `step` consists of the following fields:
+
+- **name :** `name` specifies the name of the `Function` that will be executed in this step.
+- **params :** `params` specifies an optional list of variables names and their value that Stash should use to resolve respective `Function`. If you use a variable in a `Function` specification whose value Stash cannot provide, you can pass the value of that variable using this `params` section. You have to specify the following fields for a variable:
+  - **name :** `name` of the variable.
+  - **value :** value of the variable.
+
+In the sample `Task` task specification that has been shown above, we have used `outputDir` variable in `pg-backup` function but Stash can not provide it's value. So, we have passed the value using the `params` section of the `Task` object.
+
+>Stash executes the `Functions` in the order they appear in `spec.steps` section. All the functions except the last one will be used to create `init-container` specification and the last function will be used to create `container` specification for respective backup job. This guarantee an ordered execution of the functions.
+
+#### spec.volumes
+
+`spec.volumes` specifies a list of volumes that should be mounted in the respective job created for this `Task`. In the sample we have shown above, we need to mount storage secret to the backup job. So, we have added the secret volume in `spec.volumes` section. Note that, we have used `REPOSITORY_SECRET_NAME` variable as secret name. This variable will be resolved by Stash from `Repository` specification.
```

Comment:
Where is this volume mounted? This is necessary for users to write their own Task/Function.